src/internal/errors.ts

```
@@ -197,29 +227,66 @@ function getMessage(key: string, parameters: unknown[], self: Error): string {
-    if (value.constructor && value.constructor.name) {
-      return `an instance of ${value.constructor.name}`;
+  switch (type) {
+    case "bigint": {
+      return `type bigint (${value}n)`;
     }
+    case "number": {
+      if (value === 0) {
+        return 1 / (value as number) === Number.NEGATIVE_INFINITY
+          ? "type number (-0)"
+          : "type number (0)";
+      } else if (value !== value) {
```

Biome (2.5.0)

[error] This comparison uses the same expression on both sides.

(lint/suspicious/noSelfCompare)